### DISCUSSION TITLE
Running docker image from windows

### DISCUSSION THREAD

**Discussion by tox104** [INTI-CMNB/KiBot#600] [Q&A]
Hello, could someone provide pointers on how to run images from windows.
I am total noob with Docker.
I have put this in docker_test.sh file:
```
export USER_ID=$(id -u)
export GROUP_ID=$(id -g)
export WORKDIR=RELATIVE_PATH_TO_KICAD_PROJECTS
export SUBDIR=SUBDIR_INSIDE_WORKDIR
docker run --rm -it -v /tmp/.X11-unix:/tmp/.X11-unix -e DISPLAY=$DISPLAY \
    -v $(pwd)/$WORKDIR:/home/$USER/workdir \
    --user $USER_ID:$GROUP_ID \
    --env NO_AT_BRIDGE=1 \
    --workdir="/home/$USER" \
    --volume="/etc/group:/etc/group:ro" \
    --volume="/home/$USER/.config/kicad:/home/$USER/.config/kicad:rw" \
    --volume="/home/$USER/.cache/kicad:/home/$USER/.cache/kicad:rw" \
    --volume="/etc/passwd:/etc/passwd:ro" \
    --volume="/etc/shadow:/etc/shadow:ro" \
    ghcr.io/inti-cmnb/kicad7_auto:1.6.4 /bin/bash -c "cd workdir/$SUBDIR; kibot"
```
And I am running it from git bash.
My first issue was:
`the input device is not a TTY.  If you are using mintty, try prefixing the command with 'winpty'`
so I changed this line:
`winpty docker run --rm -it -v /tmp/.X11-unix:/tmp/.X11-unix -e DISPLAY=$DISPLAY \`

But I am lost with this error:
`docker: Error response from daemon: mkdir C:\Program Files\Git\etc\group;C: Access is denied.
`

Can someone help with this?

**Comment by nopeppermint** [comment]
I had a running setup in 15min by installing docker for windows and gitlab runner for windows, cloned the kibot for gitlab example inmy gitlab account and registered my pc as gitlab runner. Now in every commit the runner on my pc is executed

**Comment by mdeweerd** [comment]
I run window's docker from cygwin as shown below.

>  mkdir C:\Program Files\Git\etc\group

This would have to be a unix-like path, it's unclear where this is coming from, but I guess this is the volume mapping "/etc/group".

```bash
docker run --rm -it -v /tmp/.X11-unix:/tmp/.X11-unix -e DISPLAY=$DISPLAY \
  -v "${CACHE_WINDIR}:${TGTHOMEDIR}/.cache" \
  -v "${SSH_WINDIR}:${TGTHOMEDIR}/.ssh" \
  ${LOCAL_KICAD_LIB_SETTING} \
  ${LOCAL_KBOTREPO} \
  ${LOCAL_KIAUTOREPO} \
  -v $WORKDIR:${TGTHOMEDIR}/workdir \
  -v "${LOCAL_KICOST_CACHE}:${TGTHOMEDIR}/kicost_cache" \
  ${KEYS_ENV} \
  --env NO_AT_BRIDGE=1 \
  ${EXTRA_OPT} \
  --workdir="${TGTHOMEDIR}" \
  ${OTHER_VOLUMES} \
  $IMAGE /bin/bash -c "$EXTRASCRIPT chmod 700 $TGTHOMEDIR/.ssh ; cd workdir/$SUBDIR ; kibo
t --version ; $ARGS" 
```

The variables are carefully set using cygpath, apparently also available with "git bash", to ensure that windows and unix-like paths are provided as needed:
```console
$ cygpath -w "c:\\windows"
C:\windows
$ cygpath -u "c:\\windows"
/c/windows
# Demonstration that "git bash" does not have /etc/group:
$ cygpath -w "/etc/group"
C:\Program Files\Git\etc\group
$ ls /etc/group
ls: cannot access '/etc/group': No such file or directory
```

Docker (on windows) itself needs windows paths, but the image needs unix paths.  You can map a volume from 'C:\path' to '/somepath', but not from '/etc/group' as that is no valid windows path.
